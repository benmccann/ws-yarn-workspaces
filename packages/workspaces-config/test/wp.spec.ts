/**
 * Created by user on 2020/4/6.
 */
import { parseWorkspaces, parseStaticPackagesPaths } from '../index';

const data = [
	"packages/@yarn-tool/*",
	"packages/*",
]

it('should parseWorkspaces', function ()
{
	let actual = parseWorkspaces(data)

	expect(actual).toStrictEqual({
		packages: [
			"packages/@yarn-tool/*",
			"packages/*",
		],
	})
	expect(actual).toMatchSnapshot();
});

it('should parseWorkspaces v2', function ()
{
	let actual = parseWorkspaces({ packages: data })

	expect(actual).toStrictEqual({
		packages: [
			"packages/@yarn-tool/*",
			"packages/*",
		],
	})
	expect(actual).toMatchSnapshot();
});

it('should parseStaticPackagesPaths', function ()
{
	let actual = parseStaticPackagesPaths(data)

	expect(actual).toStrictEqual({
		static: [],
		prefixRoot: ['packages', 'packages'],
		prefix: ['packages/@yarn-tool', 'packages'],
		prefixSub: ['@yarn-tool', ''],
		all: ['packages/@yarn-tool/*', 'packages/*'],
	})
	expect(actual).toMatchSnapshot();
});
