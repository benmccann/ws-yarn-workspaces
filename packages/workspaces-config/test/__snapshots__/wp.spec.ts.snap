// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should parseStaticPackagesPaths 1`] = `
Object {
  "all": Array [
    "packages/@yarn-tool/*",
    "packages/*",
  ],
  "prefix": Array [
    "packages/@yarn-tool",
    "packages",
  ],
  "prefixRoot": Array [
    "packages",
    "packages",
  ],
  "prefixSub": Array [
    "@yarn-tool",
    "",
  ],
  "static": Array [],
}
`;

exports[`should parseWorkspaces 1`] = `
Object {
  "packages": Array [
    "packages/@yarn-tool/*",
    "packages/*",
  ],
}
`;

exports[`should parseWorkspaces v2 1`] = `
Object {
  "packages": Array [
    "packages/@yarn-tool/*",
    "packages/*",
  ],
}
`;
